analytics_export/bigqueryClient.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const fs = require('fs');
+const tempy = require('tempy');
+const { BigQuery } = require('@google-cloud/bigquery');
+
+const { measureToTablePrefix, dimensionToTableSuffix } = require('./tableMetadata');
+
+/**
+ *  Wrapper around base bigquery client that handles all interaction logic
+ */
+class BigqueryClient {
+  constructor(dataset) {
+    this.dataset = dataset;
+  }
+
+  static async createClient(projectId, datasetId) {
+    const bqClient = new BigQuery({
+      projectId,
+    });
+
+    const dataset = bqClient.dataset(datasetId);
+    const [datasetExists] = await dataset.exists();
+
+    if (datasetExists === false) {
+      await dataset.create();
+      console.log(`Created dataset: ${datasetId}`);
+    }
+
+    return new BigqueryClient(dataset);
+  }
+
+  async createTableIfNotExists(tableName, schema, description) {
+    let table = this.dataset.table(tableName);
+    const [tableExists] = await table.exists();
+
+    if (tableExists === true) {
+      return table;
+    }
+
+    const options = {
+      schema,
+      description,
+      timePartitioning: {
+        type: 'DAY',
+        field: 'date',
+      },
+    };
+    [table] = await this.dataset.createTable(tableName, options);
+    console.log(`Created table ${table.id}`);
+    return table;
+  }
+
+  async writeData(measure, dimension, date, data, overwrite) {
+    const schema = [
+      { name: 'date', type: 'DATE', mode: 'REQUIRED' },
+      { name: 'app_name', type: 'STRING', mode: 'REQUIRED' },
+      { name: 'value', type: 'STRING', mode: 'REQUIRED' },
+    ];
+
+    if (dimension !== null) {
+      schema.push({ name: dimensionToTableSuffix[dimension], type: 'STRING', mode: 'REQUIRED' });
+    }
+
+    let tableName;
+    if (dimension !== null) {
+      tableName = `${measureToTablePrefix[measure].name}_by_${dimensionToTableSuffix[dimension]}`;
+    } else {
+      tableName = `${measureToTablePrefix[measure].name}_total`;
+    }
```

Comment:
Probably can use `const` here?

```suggestion
    const tableName = dimension ? `${measureToTablePrefix[measure].name}_by_${dimensionToTableSuffix[dimension]}` : `${measureToTablePrefix[measure].name}_total`;
```